test/extensions/filters/http/geoip/config_test.cc

```diff
@@ -0,0 +1,165 @@
+#include "envoy/extensions/filters/http/geoip/v3/geoip.pb.h"
+#include "envoy/extensions/filters/http/geoip/v3/geoip.pb.validate.h"
+
+#include "source/extensions/filters/http/geoip/config.h"
+#include "source/extensions/filters/http/geoip/geoip_filter.h"
+#include "source/extensions/filters/http/geoip/geoip_provider_config.h"
+
+#include "test/extensions/filters/http/geoip/mocks.h"
+#include "test/extensions/filters/http/geoip/utils.h"
+#include "test/mocks/server/factory_context.h"
+#include "test/test_common/registry.h"
+#include "test/test_common/test_runtime.h"
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+using ::testing::Truly;
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace Geoip {
+namespace {
+
+using GeoipFilterConfig = envoy::extensions::filters::http::geoip::v3::Geoip;
+
+TEST(GeoipFilterConfigTest, GeoipFilterDefaultValues) {
+  TestScopedRuntime scoped_runtime;
+  scoped_runtime.mergeValues({{"envoy.reloadable_features.no_extension_lookup_by_name", "false"}});
+  DummyGeoipProviderFactory dummy_factory;
+  Registry::InjectFactory<GeoipProviderFactory> registered(dummy_factory);
+  std::string filter_config_yaml = R"EOF(
+    geo_headers_to_add:
+      city: "x-geo-city"
+    provider:
+        name: "envoy.geoip_providers.dummy"
+  )EOF";
+  GeoipFilterConfig filter_config;
+  TestUtility::loadFromYaml(filter_config_yaml, filter_config);
+  NiceMock<Server::Configuration::MockFactoryContext> context;
+  EXPECT_CALL(context, messageValidationVisitor()).Times(3);
+  GeoipFilterFactory factory;
+  Http::FilterFactoryCb cb = factory.createFilterFactoryFromProto(filter_config, "geoip", context);
+  Http::MockFilterChainFactoryCallbacks filter_callback;
+  auto has_expected_config =
+      [](std::shared_ptr<Envoy::Http::StreamDecoderFilter> stream_decoder_filter) {
+        auto filter = std::static_pointer_cast<GeoipFilter>(stream_decoder_filter);
+        return !GeoipFilterPeer::useXff(*filter) &&
+               GeoipFilterPeer::xffNumTrustedHops(*filter) == 0 &&
+               GeoipFilterPeer::geoCityHeader(*filter).value() == "x-geo-city" &&
+               !GeoipFilterPeer::geoCountryHeader(*filter) &&
+               !GeoipFilterPeer::geoRegionHeader(*filter) &&
+               !GeoipFilterPeer::geoAsnHeader(*filter) &&
+               !GeoipFilterPeer::geoAnonVpnHeader(*filter) &&
+               !GeoipFilterPeer::geoAnonHostingHeader(*filter) &&
+               !GeoipFilterPeer::geoAnonTorHeader(*filter) &&
+               !GeoipFilterPeer::geoAnonProxyHeader(*filter);
+      };
+  EXPECT_CALL(filter_callback, addStreamDecoderFilter(Truly(has_expected_config)));
```

Comment:
This is *okay*, but in the event of a test failure it would have completely unhelpful output. Consider defining a bunch of matchers (I think they could be one-liners) instead of one globby function, and doing 
```
MATCHER_P(HasUseXff, expected, "") {return GeoipFilterPeer::useXff(*arg);}

EXPECT_CALL(filter_callback, addStreamDecoderFilter(AllOf(
    HasUseXff(false),
    HasXffNumTrustedHops(0),
    [etc.]
)));
```

Makes even more sense when you can also reuse this in the other `has_expected_config` case[s] below.